book/internal-services.md

```diff
@@ -0,0 +1,81 @@
+# Internal Services
+
+## Overview
+
+The Internal Services system consists of a kubernetes controller (running on an internal, private cluster) that is capable of watching and reconciling custom resources on a remote cluster.
+These custom resources describe which pipelines and parameters to use to execute internal jobs on an internal, private cluster.
+The results and outcome of the pipeline are added as an update to the custom resources. The remote cluster watches these resources to determine the outcome.
+
+## Goals
+
+* The Internal Services system attempts to enable execution of internal jobs via a polling mechanism.
+* The Internal Services system provides execution results back to the requesting cluster.
+
+## System Context
+
+The diagram below shows the interaction of the internal services controller and other services and shows the flow of custom resources
```

Comment:
Do we want to call out any of the other services by name and link to them (i.e. release service)? Or they are not _really_ a dependency so the link doesn't need to be made (in either direction)?

Reply:
mentioned Release Service